hw/mcu/stm/stm32l4xx/src/hal_flash.c

```diff
@@ -38,9 +38,12 @@ stm32_mcu_flash_erase_sector(const struct hal_flash *dev, uint32_t sector_addres
         eraseinit.TypeErase = FLASH_TYPEERASE_PAGES;
         if ((sector_address - dev->hf_base_addr) < (_FLASH_SIZE / 2)) {
```

Comment:
Is it still OK to test against `_FLASH_SIZE / 2` when there's only a single bank? Won't this stop the second half of the flash from being erased? I think maybe it would be better to make `eraseinit.Banks = FLASH_BANK_1;` before the test and then test if we're in the 2nd half and change the value to `FLASH_BANK_2` (inside the #ifdef). What do you think?

Reply:
Good call! Didn't even think of that. IO'll take a look